pkg/crypt/crypt.go

```diff
@@ -18,8 +21,14 @@ func CryptSHA512(phrase string) (string, error) {
 		return "", nil
 	}
 
-	hashSettings := "$6$" + salt
-	return crypt(phrase, hashSettings)
+	hash := sha512.New()
```

Comment:
`crypt(3)` does more than just hashing the salt and the password with sha512, see https://github.com/besser82/libxcrypt/blob/72f75aa370ae96ccd2cc44ea3cf4182d8679ffbe/lib/crypt-sha512.c#L85.

I feel quite uncomfortable reimplementing this code, because I don't want us to maintain too much crypto-related code.

Reply:
Nice blast from the past for me (c.f. https://github.com/mvo5/sha512crypt-node) - the paper what exactly sha512crypt is doing is in http://www.akkadia.org/drepper/SHA-crypt.txt but I agree with Ondrej that ideally we want to call into libc here for maximum compatibility.

Reply:
thank you @ondrejbudai maybe do you have an idea how to support mac after allow user customization with password?
https://github.com/osbuild/image-builder/pull/1283